recipe/recipe.go

```diff
@@ -36,3 +42,335 @@ type instance struct {
 type file struct {
 	Description string `bson:"description,omitempty" json:"description,omitempty"`
 }
+
+//validateInstance - checks if fields of OutputInstances are not empty for ValidateAddRecipe and ValidateAddInstance
+func (instance *Instance) validateInstance() (missingFields []string, invalidFields []string) {
+
+	if instance.DatasetID == "" {
+		missingFields = append(missingFields, "dataset-id")
```

Comment:
try and keep the way we describe a missing field in line with the spec - if a user actually tried to provide a field `dataset-id` it wouldnt work, because we're expecting the key `dataset_id` in the json

Reply:
Done in commit "optimising code by adding sub-functions and global var"